language/extensions/move-table-extension/src/lib.rs

```diff
@@ -395,12 +395,9 @@ fn native_add_box(
     let table_context = context.extensions().get::<NativeTableContext>();
     let mut table_data = table_context.table_data.borrow_mut();
 
+    // see REFERENCE SAFETY EXPLANATION in values
     let val = args.pop_back().unwrap();
-    let key = args
-        .pop_back()
-        .unwrap()
-        .value_as::<Reference>()?
-        .read_ref()?;
+    let key = unsafe { args.pop_back().unwrap().value_as::<Reference>()?.read_ref() };
```

Comment:
Wondering: can't we isolate the `unsafe` somewhere in the value impl? Really don't like to see this in user code.